mlocal/checks/basechecks.chk

```diff
@@ -224,10 +224,12 @@ fi
 ########################
 printf " checking: host os type... "
 host=
-if echo | $hstcc -E -dM - | grep -qs -e __unix__ -e __unix -e unix \
-	-e __APPLE__; then
+if echo | $hstcc -E -dM - | grep -qs -e __unix__ -e __unix -e unix; then
 	host="unix"
 fi
+if echo | $hstcc -E -dM - | grep -qs -e __APPLE__; then
```

Comment:
These changes allow us to determine if the host is macOS/Darwin rather than having that host included in the generic `unix` handling.